frameworks/Kotlin/ktor/src/main/kotlin/org/jetbrains/ktor/benchmarks/Hello.kt

```diff
@@ -37,64 +50,72 @@ fun main(args: Array<String>) {
                 call.respond(TextContent(gson.toJson(Message()), ContentType.Application.Json, HttpStatusCode.OK))
             }
             get("/db") { call ->
-                pool.connection.use { connection ->
-                    val random = ThreadLocalRandom.current()
-                    val queries = call.queries()
-                    val result = mutableListOf<World>()
-
-                    connection.prepareStatement("SELECT * FROM World WHERE id = ?", TYPE_FORWARD_ONLY, CONCUR_READ_ONLY).use { statement ->
-                        for (i in 1..(queries ?: 1)) {
-                            statement.setInt(1, random.nextInt(DbRows) + 1)
-
-                            statement.executeQuery().use { rs ->
-                                while (rs.next()) {
-                                    result += World(rs.getInt("id"), rs.getInt("randomNumber"))
+                val response = run(databaseDispatcher) {
```

Comment:
Warning: I'm not a Kotlin expert so feel free to correct me.
@martin-g Looking at [the docs](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.experimental/run.html) for `run(...)` it seems OK, because later there is the [following line](https://github.com/TechEmpower/FrameworkBenchmarks/pull/2834/files/b545caa5fbe918d0527048fde7c824b613b0d2ec#diff-4bb4adade5ee157dcce0e7cdc46088a2R78) The problem with the other Kotlin implementation was that the result of the updates was not served to the client AFTER the execution of the queries. IMO here we have the necessary wait to see if execution (of the SQL queries) will complete before serving the response to the client. Also this [seems to be the documentation](https://kotlinlang.org/docs/reference/coroutines.html) about the `coroutines`.